static/app/views/automations/components/actionNodeList.tsx

```diff
@@ -0,0 +1,85 @@
+import {Fragment} from 'react';
+import styled from '@emotion/styled';
+
+import {Select} from 'sentry/components/core/select';
+import type {
+  ActionType,
+  Integration,
+  NewAction,
+} from 'sentry/types/workflowEngine/actions';
+import {
+  ActionNodeContext,
+  actionNodesMap,
+  useActionNodeContext,
+} from 'sentry/views/automations/components/actionNodes';
+import AutomationBuilderRow from 'sentry/views/automations/components/automationBuilderRow';
+
+interface ActionNodeListProps {
+  actions: NewAction[];
+  availableActions: Array<{type: ActionType; integrations?: Integration[]}>;
+  group: string;
+  onAddRow: (type: ActionType) => void;
+  onDeleteRow: (id: number) => void;
+  placeholder: string;
+  updateAction: (index: number, data: Record<string, any>) => void;
+}
+
+export default function ActionNodeList({
+  group,
+  placeholder,
+  actions,
+  availableActions,
+  onAddRow,
+  onDeleteRow,
+  updateAction,
+}: ActionNodeListProps) {
+  const options = Array.from(actionNodesMap.entries())
+    .map(([value, {label}]) => ({
+      value,
+      label,
+    }))
+    .filter(({value}) => availableActions.some(action => action.type === value));
```

Comment:
nit:
```suggestion
const options = Array.from(actionNodesMap)
  .filter(([value]) => availableActions.some(action => action.type === value))
  .map(([value, {label}]) => ({value, label}));
```
You don't need the `.entries()` because `Array.from` will use `[Symbol.iterator]()` under the hood to determine it generically.

Then flipping the order of `map` and `filter` will _slightly_ improve performance by filtering options.

Reply:
fixed in https://github.com/getsentry/sentry/pull/90345/commits/964c7608a5a930a60bb30ce788e4fe416ecef855 !